ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java

```diff
@@ -0,0 +1,373 @@
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.powermock.reflect.Whitebox.setInternalState;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        ImageUtils.class,
+        LazyHeaders.class,
+        TextUtils.class,
+        LazyHeaders.Builder.class,
+        AttachmentPreviewFragment.class,
+        FileAttachmentUtil.class,
+        AttachmentHelper.class,
+        AttachmentHelper.AttachmentFileType.class
+
+})
+public class AttachmentPreviewFragmentTest {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    private AttachmentPreviewFragment underTest;
+    @Mock
+    private View loadingView;
+    @Mock
+    private View mRoot;
+    @Mock
+    private FileAttachmentDownloadHelper mFileAttachmentDownloadHelper;
+    @Mock
+    private DownloadAttachment mDownloadAttachment;
+    @Mock
+    private LayoutInflater inflater;
+    @Mock
+    private View view;
+    @Mock
+    private Bundle bundle;
+    @Mock
+    private FragmentHostCallback mHost;
+    @Mock
+    private Activity mActivity;
+    @Mock
+    private FragmentActivity activity;
+    @Mock
+    private View sendButton;
+    @Mock
+    private View optionsButton;
+    @Mock
+    private View attachmentDetails;
+    @Mock
+    private TextView textView;
+    @Mock
+    private TextView attachmentTime;
+    @Mock
+    private Auth auth;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        staticMock();
+        underTest = spy(new AttachmentPreviewFragment());
+
+        setInternalState(underTest, "mRoot", mRoot);
+        setInternalState(underTest, "mHost", mHost);
+        setInternalState(underTest, "mDownloadAttachment", mDownloadAttachment);
+        setInternalState(underTest, "mFileAttachmentDownloadHelper", mFileAttachmentDownloadHelper);
+        setInternalState(mHost, "mActivity", mActivity);
+
+        doReturn(activity).when(underTest).getActivity();
+    }
+
+    @Test
+    public void givenOnCreateViewThenInflaterShouldInflate() {
+        underTest.onCreateView(inflater, null, null);
+        verify(inflater).inflate(R.layout.ko__fragment_attachment_preview, null, false);
+    }
+
+    @Test
+    public void givenOnViewCreatedThenShouldSetOnClickListener() {
+        View mockedView = mock(View.class);
+        doReturn(mockedView).when(mRoot).findViewById(R.id.ko__button_exit);
+        underTest.onViewCreated(view, bundle);
+        verify(mockedView).setOnClickListener(any(View.OnClickListener.class));
+    }
+
+    @Test
+    public void givenOnActivityCreatedWhenActivityNotKayakoAttachmentPreviewActivityThenThrowIllegalStateException() {
+        expectedException.expect(Exception.class);
+        expectedException.expectMessage("This fragment is meant to be used with KayakoAttachmentPreviewActivity");
```

Comment:
✅ Good to see that exceptional cases are tested as well.

This way the code pass invalid argument/behaviors and check that appropriate exception is raised.

**No change suggested, looks good!**	  

<a id="894f7280-4faa-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />